test_install/test_install_cuda.cu

```diff
@@ -0,0 +1,367 @@
+/*******************************<GINKGO LICENSE>******************************
+Copyright (c) 2017-2020, the Ginkgo authors
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met:
+
+1. Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright
+notice, this list of conditions and the following disclaimer in the
+documentation and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+#include <ginkgo/ginkgo.hpp>
+
+
+#include <chrono>
+#include <iostream>
+#include <map>
+#include <string>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+
+// core/base/polymorphic_object.hpp
+class PolymorphicObjectTest : public gko::PolymorphicObject {};
+
+
+int main(int, char **)
+{
+    auto refExec = gko::ReferenceExecutor::create();
+    auto cudaExec = gko::CudaExecutor::create(0, refExec);
+    // core/base/abstract_factory.hpp
+    {
+        using type1 = int;
+        using type2 = double;
+        static_assert(
+            std::is_same<
+                gko::AbstractFactory<type1, type2>::abstract_product_type,
+                type1>::value,
+            "abstract_factory.hpp not included properly!");
+    }
+
+    // core/base/array.hpp
+    {
+        using type1 = int;
+        using ArrayType = gko::Array<type1>;
+        /*auto test =*/ArrayType{};
```

Comment:
Why is the `auto test =` commented out?
Commented code is usually never a good sign in production code, so I would prefer if it was removed or put back in.
This pretty much applies to all commented code in this file.

Reply:
Good point. I removed them to avoid "unused variable" warnings, but I guess we can do without them.